### ISSUE TITLE
Add traffic split info to CLI

### ISSUE THREAD

**Issue by scottcarol** [linkerd/linkerd2#3106]
`linkerd stat ts/split-name` should return data about a specific traffic split. See #3105 for context. 

Input:
`linkerd stat ts/foo-ts`

Output:
```bash
NAME     APEX    LEAF      WEIGHT     SUCCESS    RPS      LATENCY_P50   LATENCY_P95   LATENCY_P99
foo-ts   foo     foo-v1    90m        100.00%    9.0rps   1ms           1ms           1ms
foo-ts   foo     foo-v2    10m        91.25%     1.0rps   3ms           4ms           4ms
```

**Comment by grampelberg** [comment]
What does `linkerd stat ts` look like?

**Comment by scottcarol** [comment]
That’s a great question. 

To be consistent with the rest of the `linkerd stat` behavior, I’d show all trafficsplits within the `default` namespace, as we do with `linkerd stat deploy`. 

Since you can have multiple trafficsplits in the same namespace, sample input/output could look like this:

Input: `linkerd stat ts`
```bash
NAME     APEX    LEAF      WEIGHT     SUCCESS    RPS      LATENCY_P50   LATENCY_P95   LATENCY_P99
bar-ts   bar     bar-v1    33m        100.00%    1.0rps   3ms           4ms           4ms
bar-ts   bar     bar-v2    33m        100.00%    1.0rps   3ms           4ms           4ms
bar-ts   bar     bar-v3    34m        100.00%    1.0rps   3ms           4ms           4ms
foo-ts   foo     foo-v1    90m        100.00%    9.0rps   1ms           1ms           1ms
foo-ts   foo     foo-v2    10m        91.25%     1.0rps   3ms           4ms           4ms
```

Input: `linkerd stat ts -o wide`
```bash
NAME     APEX    LEAF      WEIGHT     SUCCESS    RPS      LATENCY_P50   LATENCY_P95   LATENCY_P99   READ_BYTES/SEC   WRITE_BYTES/SEC
bar-ts   bar     bar-v1    33m        100.00%    1.0rps   3ms           4ms           4ms           2988.5B/s         2871.6B/s
bar-ts   bar     bar-v2    33m        100.00%    1.0rps   3ms           4ms           4ms           2835.6B/s         2890.8B/s
bar-ts   bar     bar-v3    34m        100.00%    1.0rps   3ms           4ms           4ms           2980.0B/s         2880.2B/s
foo-ts   foo     foo-v1    90m        100.00%    9.0rps   1ms           1ms           1ms           25159.0B/s        23459.2B/s
foo-ts   foo     foo-v2    10m        91.25%     1.0rps   3ms           4ms           4ms           25159.0B/s        23459.2B/s
```
Here is the documentation that I would add to `linkerd stat`.

```bash
# Get all trafficsplits in the default namespace. 
linkerd stat ts

# Get the ts-hello trafficsplit in the default namespace. 
linkerd stat ts/ts-hello

# Get all trafficsplits in the test namespace. 
linkerd stat ts -n hello

# Get the ts-foo trafficsplit in the hello namespace. 
linkerd stat ts/ts-foo -n hello
```

Here are the flags that I would support:
```bash
      --all-namespaces              If present, returns stats across all namespaces, ignoring the "--namespace" flag
      -h, --help                    help for stat
      -n, --namespace string        Namespace of the specified resource (default "default")
      -o, --output string           Output format; one of: "table" or "json" or "wide" (default "table")
```

Flags that are not applicable:
```bash
      --from string             If present, restricts outbound stats from the specified resource name
      --from-namespace string   Sets the namespace used from lookup the "--from" resource; by default the current "--namespace" is used
      -t, --time-window string      Stat window (for example: "10s", "1m", "10m", "1h") (default "1m")
      --to string               If present, restricts outbound stats to the specified resource name
      --to-namespace string     Sets the namespace used to lookup the "--to" resource; by default the current "--namespace" is used
```

WDYT?

**Comment by grampelberg** [comment]
If we want to be super consistent, there's also TCP connections, but I don't think that actually provides much value here.

Why aren't those flags applicable? They seem to be, especially `--time-window`?

**Comment by scottcarol** [comment]
Agreed re: `--time-window`, will add. 

Re: `--to` and `--from` flags - I agree that we want to make the `ts` command as consistent as possible with the rest of `stat` but I need a little more information on how this is helpful for the user. 

If the `stat ts` command is being created to give users a big picture view of how their trafficsplit is doing, how does restricting metrics from/to a specific resource achieve that? Can't users already get drilled-down metrics information from existing `stat` functionality?

**Comment by grampelberg** [comment]
Honestly, I can't come up with a great *user* reason they'd want `--to` and `--from` off the top of my head. My primary concern is consistency. If we get `--to` and `--from` for a low cost it feels worth keeping the special logic out of the code and avoiding confusing users about what can and can't work.

If it isn't a low cost, maybe we should consider making it a separate command as it really is a separate command at that point.

**Comment by scottcarol** [comment]
I haven't dug into the queries yet, but I'll keep this in mind. The first part of the project I'm tackling is #3105, modifying the `stat` API endpoint to support `ts`. As of now I'll aim to support all existing `stat` flags. I'll build the API portion with the knowledge that we may ultimately choose to move it to its own endpoint/command.

**Comment by adleong** [comment]
`--to` and `--from` are meaningful but it's unclear how useful they are.  `--from` is probably pretty useful ("show me traffic that THESE resources are sending to the traffic split") but I can't really see `--to` being very useful ("show me traffic sent to the traffic split that ended up at THESE resources").

I don't think these flags are particularly low cost.  All of the TS stuff will likely require brand new prometheus queries so we don't get `--to`/`--from` support for free.  

My instinct would be to leave these flags out of the initial implementation and add support for them later.

**Comment by grampelberg** [comment]
If we don't end up supporting the whole flag set, we *do* need to make all those unsupported flags error out.

**Comment by scottcarol** [comment]
Yes re: erroring out unsupported flags. Once I have a proof of concept working locally, I'll explore how supporting those flags will affect the queries.

**Comment by scottcarol** [comment]
Closed by #3176
